imap/jmap_mail.c

```diff
@@ -7304,18 +7315,13 @@ static int _email_get_bodies(jmap_req_t *req,
                 is_icsbody = 1;
             }
             else {
-                struct param *param = part->disposition_params;
-                while (param) {
-                    if (!strcasecmp(param->attribute, "FILENAME") ||
-                        !strcasecmp(param->attribute, "NAME")) {
-                        size_t len = strlen(param->value);
-                        if (len > 4 && !strcasecmp(param->value + len-4, ".ICS")) {
-                            is_icsbody = 1;
-                            break;
-                        }
-                    }
-                    param = param->next;
+                struct buf buf = BUF_INITIALIZER;
+                emailbodypart_read_name(&buf, part);
+                if (buf.len > 4 &&
+                        !strcasecmp(buf_cstring(&buf) + buf.len-4, ".ICS")) {
```

Comment:
I know we use buf.len in a lot of places in the code, but should we standardize on using buf_len()?